test/functional/ex_cmds/mksession_spec.lua

```diff
@@ -47,4 +47,27 @@ describe(':mksession', function()
     command('tabnext 2')
     eq(cwd_dir .. get_pathsep() .. tab_dir, funcs.getcwd())
   end)
+
+  it('restores buffers when using tab-local working directories', function()
```

Comment:
this is almost identical to the `restores tab-local working directories` test, so that test should just be enhanced rather than creating a new one.